chroma-agent/chroma_agent/action_plugins/manage_targets.py

```diff
@@ -115,7 +115,13 @@ def get_resource_locations():
     """Parse `crm_mon -1` to identify where (if anywhere)
        resources (i.e. targets) are running."""
 
-    rc, lines_text, stderr = AgentShell.run_old(["crm_mon", "-1", "-r"])
+    try:
+        rc, lines_text, stderr = AgentShell.run_old(["crm_mon", "-1", "-r"])
+    except OSError, e:
+        # ENOENT is fine here.  Pacemaker might not be installed yet.
+        if e.errno != errno.ENOENT:
```

Comment:
It's the only _non-tampering_ reason.  Even `yum erase pacemaker` wouldn't have such a subtle effect.  It would be much more drastic.

But yes, if somebody did a `rm -f $(which crm_mon)` on their system, this change would fail that.  But there are lots of things somebody could do that would hose us in similar ways.  I think we have to draw a (reasonable) line at things we will handle.

To fix this such that the above couldn't happen (and I really did/do think it's the better way -- just not sure if it's worth the ROI) the agent would need to know the state of the agent deployment on each agent.  I.e. the agent needs to be able to ask itself "has pacemaker been installed yet?".  This is certainly not out of the realm of possibility but it is more state to track and keep sync'd/updated and a question of value.